irods/test/data_obj_test.py

```diff
@@ -1517,6 +1517,58 @@ def enter_file_into_irods( session, filename, **kw_opt ):
                 self.sess.resources.remove( resc_name )
         self.assertIs( default_XML_parser(), current_XML_parser() )
 
+
+    def test_data_open_on_named_resource__243(self):
+        s = self.sess
+        data = s.data_objects
+        home = helpers.home_collection(s)
+        data_name = unique_name(my_function_name(),datetime.now(),'data')
+        resc_name = unique_name(my_function_name(),datetime.now(),'resc')
+        with self.create_simple_resc(resc_name) as resc:
+            data_path = '{home}/{data_name}'.format(**locals())
+            try:
+                with data.open(data_path,'w',**{kw.RESC_NAME_KW:resc}) as f:
+                    f.write(b'abc')
+                d = data.get(data_path)
+                self.assertEqual(len(d.replicas),1)
+                self.assertEqual(d.replicas[0].resource_name, resc)
```

Comment:
Not sure how much of this is testing the client versus testing the server, but maybe you could target a non-root resource with this keyword and ensure that the right error message/code is returned? Should give you a `DIRECT_CHILD_ACCESS`, I think.

Reply:
I guess that would be valuable, yes.  Not to mention DIRECT_CHILD_ACCESS seems ... not to be defined  in `irods.exceptions`. That would need fixing too.